### ISSUE TITLE
Upload Skript to SpigotMC or other plugin distribution sites

### ISSUE THREAD

**Issue by XfedeX** [SkriptLang/Skript#3678]
Can anyone please update https://dev.bukkit.org/projects/skript?
Some server host only accepts plugins from Bukkit or SpigotMc **WITHOUT EXTERNAL DOWNLOAD**.
Thanks.

**Comment by Romitou** [comment]
See https://github.com/SkriptLang/Skript/issues/3550#issuecomment-728813877

**Comment by XfedeX** [comment]
@Romitou Could anyone please make another project? Around 60 000 players use Aternos every day: Skript would become really bigger.

**Comment by Romitou** [comment]
GitHub includes a full and powerful API. Why doesn't your host use it? I would advise you to change your hosting provider if they impose constraints on you because they don't want to adapt.

**Comment by XfedeX** [comment]
Cant you just upload it?

**Comment by WeeskyBDW** [comment]
You didn't have acces to a FTP/SFTP connexion ? For upload it manually

**Comment by Njol** [comment]
I could also give someone access to the original Bukkit plugin page, if desired.

**Comment by Olyno** [comment]
Hi @Njol, nice to see you here again. If it doesn't bother you, we could give the access to the Bukkit page of Skript directly to the owner of the SkriptLang's organization, @bensku.
I think we can add contributors to an uploaded plugin (but how, I don't know ahah, I never tried).

Anyway, it's really nice to you to propose that to us 🙂

**Comment by Moderocky** [comment]
It wouldn’t actually help with this, though.
You can’t upload a plugin to Bukkit that doesn’t work on Bukkit.
Skript could be added to Spigot but never has been because *reasons*?

**Comment by XfedeX** [comment]
@Moderocky I said the reasons, so can anyone upload it to spigotMC? I uploaded by myself but it just disappeared after 1 day... i think someone deleted it...

**Comment by Moderocky** [comment]
> i think someone deleted it...
Yes, it will be deleted if you try because you didn't create it and Spigot doesn't allow people to upload resources they didn't create.
> so can anyone upload it to spigotMC?
Bensku could, but hasn't. I don't know the reason why.

**Comment by bensku** [comment]
I actually planned to publish on SpigotMC a few years ago, but got busy with other things. I'm currently not very active with Skript development, but I would certainly help if anyone else in core team wanted to publish Skript on SpigotMC. Same goes for BukkitDev, altough @Moderocky is correct: Skript requires a Spigot API implementation.

In any case, publishing to SpigotMC would might not even help with Aternos. As far as I know, they require plugins to be whitelisted before they can be installed. Skript has been built with the assumption that scripts are trusted not to do anything malicious. Allowing it on whitelist of 'safe' plugins could (depending on threat model and other security measures) be a security risk.

**Comment by ShaneBeee** [comment]
My problem with this whole situation, is the fact that Aternos is dictating how/where we need to upload our resource.

1) We have to have it on SpigotMC.org or DBO
2) it has to be a DIRECT download (ie: we can't have the link forward to the releases page and/or specific build on GitHub)

It really gets under my skin that Aternos refuses to use the GitHub API, with the excuse "spigot has information specific to Minecraft whereas GitHub does not" ... the only thing SpigotMC offers in their download API is the MC version (which is often missing and/or incorrect for a lot of plugins), this information doesn't matter.
Personally on my plugins, I have the spigot download links direct to the specific build on GitHub, this way anyone downloading said resource is made sure they see the changelog.
Quite often people download builds from spigot without knowing what has changed, then complain about stuff cause they didn't bother reading the changelog. This makes sure they actually see the changelog.

It also bothers that myself (and many many other devs I know) have been harassed by Aternos users/staff because our resources are "not available for download"
I've seen devs get 1 star reviews on Spigot (from Aternos users) because they don't offer a direct download.
I myself have received 1 star reviews because my resources don't have a direct download (I had these reviews removed, because the review had nothing to do with the resource itself, but rather the anger from a user because they "couldn't get it")
Why should devs get harassed because a user choses a host which doesn't offer their plugin?!?! How is that the devs fault Aternos **REFUSES** to use the GitHub API?

> Around 60 000 players use Aternos every day: Skript would become really bigger.

Player count makes no difference to the size of Skript users. Its servers running it that matters. 

It puts such a bad taste in my mouth that Aternos refuses to use the GitHub API, but then has the nerve to harass people because we don't bend over backwards for them... a server host.

According to bStats, Skript is used on over 6000 servers. Its well know that a lot of servers turn off bStats, so this number is most likely at least doubled. On top of that, it makes no difference if we have 6000 users or 50000 users. 

Me personally, due to how I've been treated by Aternos users/staff, I won't be putting the resource up on SpigotMC. If I was going to, I would follow suit with how I do on other resources (download link would lead to the GitHub release of said build so that the users HAVE to read the changelog) ... but then we would still be getting these issues because Aternos "can't" download it. 

I don't think Skript, nor any resource/dev, should have to bend over backwards for **ONE** server host, because they feel entitled to dictate how devs should operate their resources/downloads.

**Comment by Moderocky** [comment]
At this point you might consider waiting for PaperMC’s hangar to be released and posting it there as the official place (if you put it anywhere at all) - I know you’ve been adding a lot of paper-supporting syntaxes recently, and with paper now having the majority share it would make more sense in the long run.

I agree with you Shane about not wanting to pander to these people, but also having an official page helps to control the ‘image’ so to speak. Somebody uploads a copy of Skript to spigot every week and having something official out there (even if it linked back to GitHub) would at least help to stop that from happening.

It would also help with the people rocking up asking for help using the ancient version they found on Bukkit because they didn’t look far enough.

That said, a spigot page would likely be reduced to spam reviews asking for help, spam reviews from spigot developers who feel emasculated by Skript’s existence and no good news.

I think there would be pros and cons.

**Comment by ShaneBeee** [comment]
I agree with the reviews thing. I feel like the majority of the reviews on spigot would be "Skript is **** ... learn Java"

I do know that md_5 will not tolerate the hate towards Skript. I have seen him lock a few threads that slander Skript, with some lovely words telling the devs they're basically intimidated ... etc. So I feel like they would handle unnecessary reviews in a nice manner. 

That said, I would be ok adding Skript to SpigotMC, because as you (Moderocky) stated it would help with the image, in a sense showing Skript as a "real" resource so to speak. 
If I did so, I would have the download link lead right back to the GitHub release, thus making sure whomever is downloading, is actually seeing the changelog.
That in itself does not solve the Aternos issue, but as I stated, I will not cater to their demands.
